sdk/core/azure-core/src/main/java/com/azure/android/core/internal/util/CancellationTokenImpl.java

```diff
@@ -60,6 +76,38 @@ public void cancel() {
         }
     }
 
+    @Override
+    public void start() {
+        if (timeout.equals(Duration.ZERO) || isStarted.get()) {
+            return;
+        }
+
+        UUID expectedRunId = runId;
+        Handler handler = new Handler();
```

Comment:
It looks like this Handler is deprecated https://developer.android.com/reference/android/os/Handler#Handler(), we should use the application main looper or executor.